classes/Course.php

```diff
@@ -2224,16 +2221,20 @@ public function tutor_add_gutenberg_author( $data, $postarr ) {
 	 * @return void
 	 */
 	public function attach_product_with_course( $post_ID, $post_data ) {
+		/**
+		 * For native monetization, just return
+		 * No need to attach anything.
+		 */
+		if ( tutor_utils()->is_monetize_by_tutor() ) {
```

Comment:
This method adds a redundant database call. First checking
`tutor_utils()->is_monetize_by_tutor()` database called
Then again: 
`$monetize_by = tutor_utils()->get_option( 'monetize_by' );` database called for the same thing

It's better to reinstate the code snippet

```
		$monetize_by = tutor_utils()->get_option( 'monetize_by' );
		$product_id  = Input::post( '_tutor_course_product_id', 0, Input::TYPE_INT );

		if ( Ecommerce::MONETIZE_BY === $monetize_by ) {
			return;
		}
```